core/jobs/batch_jobs/reject_invalid_suggestion_and_delete_invalid_translation_jobs.py

```diff
@@ -0,0 +1,596 @@
+# coding: utf-8
+#
+# Copyright 2024 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs for rejecting translation suggestions for already translated
+content and deleting translations for invalid content ID."""
+
+from __future__ import annotations
+
+from core import feconf
+from core.domain import exp_fetchers
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+from typing import Dict, Iterable, List, Optional, Sequence, Tuple, Union
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import datastore_services
+    from mypy_imports import exp_models
+    from mypy_imports import opportunity_models
+    from mypy_imports import suggestion_models
+    from mypy_imports import translation_models
+
+(
+    exp_models,
+    opportunity_models,
+    suggestion_models,
+    translation_models) = models.Registry.import_models([
+        models.Names.EXPLORATION,
+        models.Names.OPPORTUNITY,
+        models.Names.SUGGESTION,
+        models.Names.TRANSLATION])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class RejectTranslationSuggestionsForTranslatedContentsJob(base_jobs.JobBase):
```

Comment:
Do you have an audit version of this job? You should always run an audit version of the job that doesn't do any put(), and have it provide enough debugging output to verify that what you're doing is correct, before running the non-audit version.

Try to make the audit and non-audit jobs share as much code as possible (have them reuse functions) -- don't duplicate logic between the jobs.

Reply:
done